### DISCUSSION TITLE
Insert data is float vector but get vector float32 when query

### DISCUSSION THREAD

**Discussion by HoangSang1510** [milvus-io/milvus#32224] [Q&A and General discussion] [answered]
Hello, I have create a collection with field 'vector':
schema.add_field(field_name="vector", datatype=DataType.FLOAT_VECTOR, dim=5)

when i insert data, my 'vector' is list of float like:
[-0.48851287364959717, -0.22706535458564758, 0.0740056037902832, -0.5651883482933044, -0.04749017208814621]

but when i query it, output vector i get is type float32, like:
[-0.48851287, -0.22706535, 0.07400560, -0.56518834, -0.04749017]

Am i doing wrong somewhere, thanks you

**Comment by yiwangdr** [comment, marked as answer]
Milvus supports storing vectors as Float32 arrays. A Float32 value has a precision of ~7 decimal digits. Even with a Float64 value, such as 1.3476964684980388, Milvus stores it as 1.347696. Therefore, when you retrieve such a vector from Milvus, the precision of the Float64 value is lost.

**Comment by xiaofan-luan** [comment]
And from my previous experience, store embeddings in fp64 is not a real option?

usually people works with fp32, fp16, int8 or even binary embeddings
